File: Data Structure & Algorithm/Models/Sort/Bubble Sort/Bubble Sort.py
from typing import List


def bubbleSort(nums: List[int]):
    for i in range(len(nums), 0, -1):
        for j in range(0, i - 1):
            if nums[j] > nums[j + 1]:
                temp = nums[j]
                nums[j] = nums[j + 1]
                nums[j + 1] = temp


# test
array = [10, 1, 3, 2, 9, 1, 5, 6]
bubbleSort(array)
print(array)

File: Data Structure & Algorithm [Old]/Template/Sort/Bubble Sort/Bubble Sort.go
package main

import "fmt"

func bubbleSort(nums []int) {
	for i := len(nums); i > 0; i-- {
		for j := 0; j < i-1; j++ {
			if nums[j] > nums[j+1] {
				temp := nums[j]
				nums[j] = nums[j+1]
				nums[j+1] = temp
			}
		}
	}
}

// test
func main() {
	array := []int{10, 1, 3, 2, 9, 1, 5, 6}
	bubbleSort(array)
	fmt.Println(array)
}
